### PR TITLE
Create update-xunit.md

### PR DESCRIPTION
**Pull request by ViktorHofer**
cc @dotnet/source-build-internal for the source-build-external part in the doc.

### CODE REVIEW

File: Documentation/update-xunit.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+This document aims to establish the necessary actions to update the xunit version in Arcade which then gets propagated into consuming repositories.
+
+1. For security reasons, xunit packages need to be manually mirrored from nuget.org to the dotnet-public AzDO feed. [See the instructions](https://github.com/dotnet/arcade/blob/main/Documentation/MirroringPackages.md). Mirror the following packages: `xunit,xunit.runner.reporters,xunit.runner.utility,xunit.runner.console,xunit.runner.visualstudio` with version `latest`.
```
Comment (akoeplinger) [inline Documentation/update-xunit.md:3]: nit: this isn't xunit specific

```suggestion
1. For security reasons, nuget packages need to be manually mirrored from nuget.org to the dotnet-public AzDO feed. [See the instructions](https://github.com/dotnet/arcade/blob/main/Documentation/MirroringPackages.md). Mirror the following packages: `xunit,xunit.runner.reporters,xunit.runner.utility,xunit.runner.console,xunit.runner.visualstudio` with version `latest`.
```

File: Documentation/update-xunit.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+2. Update `XUnitVersion`, `XUnitAnalyzersVersion` and `XUnitRunnerVisualStudioVersion` properties in [Arcade SDK's DefaultVersions.props](https://github.com/dotnet/arcade/blob/074491b2a86631659e28a712439befcea29041ee/src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props#L84) to the desired values. Make sure to use a coherent version of `xunit.analyzers`.
```
Comment (akoeplinger) [inline Documentation/update-xunit.md:4]: I think you can use relative URLs instead of absolute ones

**Review by akoeplinger — APPROVED** [review]

File: Documentation/update-xunit.md (hunk 2 above)
Comment (ViktorHofer) [reply Documentation/update-xunit.md]: done

**Review by akoeplinger — DISMISSED** [review]

**Review by akoeplinger — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]

**Review by MichaelSimons — APPROVED** [review]

**Comment by ViktorHofer** [comment]
@agocke @mmitche PTAL

**Review by agocke — APPROVED** [review]

**Pull request merged**